### PR TITLE
Make Docker plugin portable so that Windows is supported

### LINKED ISSUE #760: sbt docker:publishLocal fails on Windows
Running "sbt docker:publishLocal" fails with this stack trace on Windows (works fine on Linux):

```scala
java.util.regex.PatternSyntaxException: Unexpected internal error near index 1
\
 ^
        at java.util.regex.Pattern.error(Unknown Source)
        at java.util.regex.Pattern.compile(Unknown Source)
        at java.util.regex.Pattern.<init>(Unknown Source)
        at java.util.regex.Pattern.compile(Unknown Source)
        at java.lang.String.split(Unknown Source)
        at java.lang.String.split(Unknown Source)
        at com.typesafe.sbt.packager.docker.DockerPlugin$.com$typesafe$sbt$packager$docker$DockerPlugin$$makeAdd(DockerPlugin.scala:152)
        at com.typesafe.sbt.packager.docker.DockerPlugin$$anonfun$projectSettings$8.apply(DockerPlugin.scala:79)
        at com.typesafe.sbt.packager.docker.DockerPlugin$$anonfun$projectSettings$8.apply(DockerPlugin.scala:70)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:235)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
[error] (*:dockerCommands) java.util.regex.PatternSyntaxException: Unexpected internal error near index 1
```

I guess it is related to the Windows directory separator "\".

### PR DESCRIPTION
**Pull request by mkotsbak**
Closes #760.

### CODE REVIEW

**Comment by muuki88** [comment]
Thanks for your pull request. Can you fix the compile error and squash the commits?

```
[error] /home/travis/build/sbt/sbt-native-packager/src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala:64: not found: value File
[error]   val SeparatorChar = File.separatorChar
[error] 
```

**Comment by mkotsbak** [comment]
Ah sorry, forgot that addition. Pushed a new commit now.

**Comment by muuki88** [comment]
@fiadliel do you have a windows computer at hand to verify it works?

**Comment by mkotsbak** [comment]
I'll add that I have only tested this on Linux myself (I have troubles running Docker-Virtualbox inside Virtualbox), the error was reported to me by another.

**Comment by muuki88** [comment]
@mkotsbak no problem :) Most of the contributors use Mac/Linux, so this is only a safety choice to have it confirmed by some systems ( we all know the "it works on my machine ;) ), as the roundtrip time for windows fixes is usually longer.

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala [hunk 1]
```diff
@@ -19,6 +19,7 @@ import packager.Keys._
 import linux.LinuxPlugin.autoImport.{ daemonUser, defaultLinuxInstallLocation }
 import universal.UniversalPlugin.autoImport.stage
 import SbtNativePackager.{ Universal, Linux }
+import java.io.File
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/nepomukseiler/sbt-native-packager/file/2356068302/issues/source?bid=2040925&fileBranchId=3219012#l22)

**Comment by fiadliel** [comment]
I don't, but I'll try and rope in a friend to try the changes. (Windows has been quite the endangered species in any place I've worked for, for a long time).

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala (hunk 1 above)
Comment (mkotsbak) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala]: Fixed it now.

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala [hunk 2]
```diff
@@ -2,6 +2,7 @@ package com.typesafe.sbt
 package packager
 package docker
 
+import java.io.File
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/nepomukseiler/sbt-native-packager/file/2358107371/issues/source?bid=2040925&fileBranchId=3219012#l5)

Comment (mkotsbak) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala]: Still? I don't understand what it wants. Maybe it doesn't support Java packages? Or should it be above the package declarations?

Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala]: I think this is an issue with the weird `package` notation used in native-packager. At some point I will fix this. For now ignore this warning. Thanks for paying attention :)

**Comment by muuki88** [comment]
I will merge this after the `1.1.0` release.

**Pull request merged**